src/config_default_credentials.rs

```diff
@@ -103,5 +106,31 @@ struct RefreshRequest<'a> {
     refresh_token: &'a str,
 }
 
+#[cfg(any(target_os = "linux", target_os = "macos"))]
+fn get_user_credentials_path() -> Result<PathBuf, Error> {
+    let mut home = home::home_dir().ok_or(Error::Str("home directory not found"));
+    home.push(CONFIG_DIR);
+    home.push(USER_CREDENTIALS_PATH);
+    Ok(home)
+}
+
+#[cfg(target_os = "windows")]
+fn get_user_credentials_path() -> Result<PathBuf, Error> {
+    let app_data = std::env::var(ENV_APPDATA)
+        .map_err(|_| Error::Str("APPDATA environment variable not found"))?;
+    let mut home = PathBuf::from(app_data);
+    if !home.exists() {
+        return Err(Error::Str("APPDATA directory not found"));
+    }
+    home.push(USER_CREDENTIALS_PATH);
+    Ok(home)
```

Comment:
In light of the above, would like to spell this as follows:

```rust
match home.exists() {
    true => Ok(home)
    false => Err(Error::Str("APPDATA directory not found")),
}
```

Reply:
done.